### PR TITLE
feat(doc): add migration section

### PR DESCRIPTION
**Pull request by tkrugg**
added a section list the changes and steps to migrate from v0 to v1.
This section will later go inside the official doc, let put it here for the time being.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
 
 - `yarn dev` - runs webpack and node dev server
 - `yarn build:dev` - runs rollup in watch mode - livereload if you do changes to the insights library
+
+## Migrating from v0 to v1
+
+### `init` method signature has changed
+- `applicationID` is now called `appId`, to stay consistent with our [other js libraries](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/?language=javascript#previous-usage).
```
Comment (francoischalifour) [inline README.md]:
```suggestion
- `applicationID` is now called `appId` to stay consistent with our [other JavaScript libraries](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/?language=javascript#previous-usage).
```

File: README.md [hunk 2]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+
+### `initSearch` method has been removed
+This method was previously used to setup pass getQueryID helper. Now you need to explicitly call this helper
```
Comment (francoischalifour) [inline README.md]:
```suggestion
This method was previously used to setup pass `getQueryID` helper. Now you need to explicitly call this helper
```

> to setup pass

There might be a typo here?

File: README.md [hunk 3]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+and pass the result to methods that require it (namely `clickededObjectIDsAfterSearch` and `convertedObjectIDsAfterSearch`)
+
+### `click` and `convert` method have been renamed and their signatures changed to reflect the different use cases covered by the insights client
```
Comment (francoischalifour) [inline README.md]:
```suggestion
### `click` and `convert` methods have been renamed and their signatures have changed to reflect the different use cases covered by the insights client
```

File: README.md [hunk 4]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+
+To make it clear they are intended to be called in the context of a search
```
Comment (francoischalifour) [inline README.md]:
```suggestion
To make it clear, they are intended to be called in the context of a search
```

File: README.md [hunk 5]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
 - `yarn build:dev` - runs rollup in watch mode - livereload if you do changes to the insights library
+
+## Migrating from v0 to v1
+
+### `init` method signature has changed
+- `applicationID` is now called `appId`, to stay consistent with our [other js libraries](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/?language=javascript#previous-usage).
+
+### `initSearch` method has been removed
+This method was previously used to setup pass getQueryID helper. Now you need to explicitly call this helper
+and pass the result to methods that require it (namely `clickededObjectIDsAfterSearch` and `convertedObjectIDsAfterSearch`)
```
Comment (francoischalifour) [inline README.md]:
```suggestion
and pass the result to methods that require it (namely `clickedObjectIDsAfterSearch` and `convertedObjectIDsAfterSearch`)
``

File: README.md [hunk 6]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+
+### `click` and `convert` method have been renamed and their signatures changed to reflect the different use cases covered by the insights client
+
+To make it clear they are intended to be called in the context of a search
+- `click` is now `clickededObjectIDsAfterSearch`
+- `convert` is now `convertedObjectIDsAfterSearch`
+
+The signatures have also changed:
+
+* On `clickededObjectIDsAfterSearch `
```
Comment (francoischalifour) [inline README.md]:
```suggestion
* On `clickedObjectIDsAfterSearch`
```

File: README.md [hunk 7]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+  - `eventName : string` is now required
+  - `index : string` is now required
+  - `objectID : number | string` is now `objectIDs : Array<number | string>`
+  - `queryID : string` is now required, use the getQueryID helper documented
```
Comment (francoischalifour) [inline README.md]:
```suggestion
  - `queryID : string` is now required, use the `getQueryID` helper documented
```

File: README.md [hunk 8]
```diff
@@ -195,3 +195,33 @@ To run all examples and play around with the code you have to run two separate c
...
+### `click` and `convert` method have been renamed and their signatures changed to reflect the different use cases covered by the insights client
+
+To make it clear they are intended to be called in the context of a search
+- `click` is now `clickededObjectIDsAfterSearch`
+- `convert` is now `convertedObjectIDsAfterSearch`
+
+The signatures have also changed:
+
+* On `clickededObjectIDsAfterSearch `
+  - `eventName : string` is now required
```
Comment (francoischalifour) [inline README.md]: There shouldn't be any space before the colon: `eventName: string`.

File: README.md (hunk 4 above)
Comment (tkrugg) [reply README.md]: I don't think that calls for a `,` 
Maybe I should rephrase that into 
> To make it clear **that** they are intended to be called in the ...
any thoughts?

Comment (francoischalifour) [reply README.md]: That's better indeed!

**Pull request merged**